src/actions/airtable/airtable.ts

```diff
@@ -72,26 +71,186 @@ export class AirtableAction extends Hub.Action {
     return new Hub.ActionResponse(response)
   }
 
-  async form() {
+  async checkBaseList(request: Hub.ActionRequest) {
+    if (request.params.state_json) {
+      const stateJson = JSON.parse(request.params.state_json)
+      const response = await this.refreshTokens(stateJson.tokens.refresh_token)
+      return gaxios.request({
+        method: "GET",
+        url: "https://api.airtable.com/v0/meta/bases",
+        headers: {
+          Authorization: `Bearer ${(response.data as any).access_token}`,
+        },
+      }).catch((err) => {
+        winston.error(JSON.stringify(err))
+        throw "Error listing bases, oauth probably bad."
+      })
+    } else {
+      return null
```

Comment:
We should consider logging the fact that json state is missing, likely with message "Missing json state -- OAuth probably failed"